packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md

```diff
@@ -0,0 +1,90 @@
+### 1: Start password recovery
+
+To begin the password recovery flow, the user must
+
+1. Click the **Forgot Password?** link on the sign-in page.
+2. Enter their **Email or Username** in the box and click **Next**.
+3. Choose **Email** as the authenticator they want to use for password recovery and click **Submit**.
+
+Okta then sends the user an email that matches the Forgot Password template that was altered earlier and the widget tells them either to click the link in the email or enter the OTP to continue.
+
+<div class="common-image-format">
+
+![Screenshot of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password Recovery Email")
+
+</div>
+
+The **Reset Password** link in the email includes the `otp` and `request.relayState` variables as query parameters back to the application. For example,
+
+`http://localhost:8080/magic-link/callback?otp=${oneTimePassword}&state=${request.relayState}` becomes `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler method that takes the `otp` and `state` parameters in the query string and passes them as session parameters to a page that contains the Sign-In Widget. First, retrieve the current `IDXClientContext` object from the browser session and check that `state` has a value.
+
+If `state` is `null` or the browser context object is `null` (because the user is clicking the magic link in a different browser), the widget will advise the user to return to the original tab in the browser where they requested a password reset and enter the OTP to proceed.
+
+```java
+@GetMapping("/magic-link/callback")
+public ModelAndView handleMagicLinkCallback(
+   HttpServletRequest request,
+   @RequestParam(name = "state") String state,
+   @RequestParam(name = "otp") String otp,
+   HttpSession session) throws MalformedURLException
+{
+  if (session.getAttribute(IDX_CLIENT_CONTEXT) == null) {
+      try {
+         idxClientContext = idxAuthenticationWrapper.getClientContext();
+      }
+      catch (ProcessingException e) {
+         // return error details
+      }
+      session.setAttribute(IDX_CLIENT_CONTEXT, idxClientContext);
+  }
+
+  if (idxClientContext == null) {
+     // return error details
+  }
+
+  if (state == null) {
+      return new ModelAndView("redirect:" + oktaOAuth2Properties.getRedirectUri());
+  }
+
+```
+
+If both values are valid, collate all the values required by the widget and pass them to the default Redirect URI for your app integration.
```

Comment:
Can we mention how setting these values here relates to the following javascript just so folks can understand how the OTP and other values pass to the widget: 

      /*<![CDATA[*/

      var config = {};

      config.baseUrl = /*[[${oktaBaseUrl}]]*/ 'https://{yourOktaDomain}';
      config.clientId = /*[[${oktaClientId}]]*/ '{clientId}';
      config.otp = /*[[${otp}]]*/ '{otp}';
      config.redirectUri = /*[[${redirectUri}]]*/ '{redirectUri}';
      config.useInteractionCodeFlow = true;
      config.interactionHandle = /*[[${interactionHandle}]]*/ '{interactionHandle}';
      config.codeChallenge = /*[[${codeChallenge}]]*/ '{codeChallenge}';
      config.codeChallengeMethod = /*[[${codeChallengeMethod}]]*/ '{codeChallengeMethod}';
      config.redirect = 'always';
      config.authParams = {
          issuer: /*[[${issuerUri}]]*/ '{issuerUri}',
          pkce: true,
          state: /*[[${state}]]*/ '{state}' || false,
          nonce: /*[[${nonce}]]*/ '{nonce}',
          scopes: /*[[${scopes}]]*/ '[scopes]',
      };

      new OktaSignIn(config).showSignInAndRedirect(
          { el: '#sign-in-widget' },
          function (res) {}
      );
      /*]]>*/

Reply:
I've added a note quoting this code and how it passes the values into the widget.